### PR TITLE
Added excerpt values to search indexing

### LINKED ISSUE #1636: Search tries to index Object instead of ID from CLI with custom document type.
If I understand correctly:

- @stemod has created a new structure type, based on the Snippet document type, which has a Category field.
- When saving in the admin, the category is correctly updated.
- When reindexing the search from the CLI, it throws an error because the relation is an object and not an ID.

### PR DESCRIPTION
**Pull request by wachterjohannes**
__tasks:__

- [x] test coverage
- [ ] gather feedback for my changes
- [x] submit changes to the documentation (see massive search docs)

__related PRs:__

* https://github.com/massiveart/MassiveSearchBundle/pull/60

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | fixes #1636 
| BC breaks        | CategoryList id array instead of complete data => see UPGRADE
| Documentation PR | none

### CODE REVIEW

**Comment by wachterjohannes** [comment]
problem with new tag (a name will be returned) => found in media edit

File: UPGRADE.md [hunk 1]
```diff
@@ -2,6 +2,16 @@
...
 
+### Search
+
+To index multiple fields (and `category_list` content-type) you have to add the attribute `type="array"` to the
+`sulu.search.field` tag. The `tag_list` content-type has its own search-field type `tags`
+(`<tag name="sulu.search.field" type="tags"/>`).
+
+### Category Content-Type
+
+The category now content-type converts the selected ids into category data only for website rendering.
```
Comment (danrot) [inline UPGRADE.md]: The `now` is misplaced, please just put it at the end of the sentence.

File: composer.json [hunk 2]
```diff
@@ -23,7 +23,7 @@
         "guzzle/guzzle": "3.*",
         "imagine/imagine": "~0.6.1",
         "liip/theme-bundle": "1.3.*",
-        "massive/search-bundle": "dev-develop",
+        "massive/search-bundle": "dev-array_support",
```
Comment (danrot) [inline composer.json]: Don't forget to change back once merged.

File: src/Sulu/Bundle/CategoryBundle/Content/Types/CategoryList.php [hunk 3]
```diff
@@ -69,16 +69,26 @@ protected function setData($data, PropertyInterface $property)
...
         $categoryIds = $node->getPropertyValueWithDefault($property->getName(), []);
+        $this->setData($categoryIds, $property);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getContentData(PropertyInterface $property)
+    {
+        $categoryIds = parent::getContentData($property);
```
Comment (danrot) [inline src/Sulu/Bundle/CategoryBundle/Content/Types/CategoryList.php]: I wouldn't do this, it's just introducing more coupling to the parent class.

File: UPGRADE.md [hunk 4]
```diff
@@ -2,6 +2,16 @@
 
 ## dev-develop
 
+### Search
+
+To index multiple fields (and `category_list` content-type) you have to add the attribute `type="array"` to the
+`sulu.search.field` tag. The `tag_list` content-type has its own search-field type `tags`
+(`<tag name="sulu.search.field" type="tags"/>`).
```
Comment (danrot) [inline UPGRADE.md]: Why does the tag need its own type, and the categories don't?

Comment (wachterjohannes) [inline UPGRADE.md]: the categories returns already an id array ... but we handle the tags a lot with names therefor we have to convert them ...

**Pull request merged**